docs/data/toolpad/reference/api/create-data-provider.md

```diff
@@ -0,0 +1,91 @@
+# createDataProvider API
+
+<p class="description">Define a backend to load server-side collections.</p>
+
+## Import
+
+```jsx
+import { createDataProvider } from '@mui/toolpad/server';
+```
+
+## Description
+
+```jsx
+import { createDataProvider } from '@mui/toolpad-core/server';
+import DATA from './movies.json';
+
+export default createDataProvider({
+  async getRecords({ paginationModel: { start = 0, pageSize } }) {
+    const records = DATA.slice(start, start + pageSize);
+    return { records, totalCount: DATA.length };
+  },
+});
+```
+
+Data providers expose collections to the Toolpad frontend. They are server-side data structures that abstract the loading and manipulation of a backend collection of records of similar shape. They can be directly connected to data grids to display the underlying data.
+
+## Parameters
+
+- `config` An object describing the data provider capabilities
```

Comment:
```suggestion
- `config: DataProviderConfig` An object describing the data provider capabilities
```

Maybe even with a link to the `DataProviderConfig` would give a much better experience.